azure-devops/projects/npm-packages/io-functions-commons.tf

```
@@ -0,0 +1,171 @@
+variable "io-functions-commons" {
+  default = {
+    repository = {
+      organization   = "pagopa"
+      name           = "io-functions-commons"
+      branch_name    = "master"
+      pipelines_path = ".devops"
+    }
+    pipeline = {
+      cache_version_id = "v3"
+    }
+  }
+}
+
+#
+# Code Review pipeline
+#
+
+# Define code review pipeline
+resource "azuredevops_build_definition" "io-functions-commons-code-review" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-pr, azuredevops_project.project]
+
+  project_id = azuredevops_project.project.id
+  name       = "${var.io-functions-commons.repository.name}.code-review"
+  path       = "\\${var.io-functions-commons.repository.name}"
+
+  pull_request_trigger {
+    initial_branch = var.io-functions-commons.repository.branch_name
+    forks {
+      enabled       = false
+      share_secrets = false
+    }
+    override {
+      auto_cancel = false
+      branch_filter {
+        include = [var.io-functions-commons.repository.branch_name]
+      }
+      path_filter {
+        exclude = []
+        include = []
+      }
+    }
+  }
+
+  repository {
+    repo_type             = "GitHub"
+    repo_id               = "${var.io-functions-commons.repository.organization}/${var.io-functions-commons.repository.name}"
+    branch_name           = var.io-functions-commons.repository.branch_name
+    yml_path              = "${var.io-functions-commons.repository.pipelines_path}/code-review-pipelines.yml"
+    service_connection_id = azuredevops_serviceendpoint_github.io-azure-devops-github-pr.id
+  }
+
+  variable {
+    name         = "DANGER_GITHUB_API_TOKEN"
+    secret_value = data.azurerm_key_vault_secret.key_vault_secret["DANGER-GITHUB-API-TOKEN"].value
+    is_secret    = true
```

```suggestion
    is_secret    = true
    allow_override = false
```

add allow_override = false for all variables

---

For example give only deploy permission to a user, the user can't modify variables to avoid deploy on unexpected app service